public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -363,10 +372,26 @@ export class PanelModel {
     return clone;
   }
 
+  getDatasourceConfig() {
+    return {
+      getTransformations: () => this.transformations,
+      getFieldOverrideOptions: () => {
+        if (!this.plugin) {
+          return undefined;
+        }
+
+        return {
+          fieldOptions: this.options.fieldOptions,
+          replaceVariables: this.replaceVariables,
+          custom: this.plugin.customFieldConfigs,
+          theme: config.theme,
+        };
+      },
+    };
+  }
```

Comment:
Another approach that I was thinking of is for PanelModel to implement PanelConfig , no sure which is best